src/gtpu/encap.c

```diff
@@ -835,18 +850,22 @@ static int gtp5g_fwd_skb_encap(struct sk_buff *skb, struct net_device *dev,
 
     pdr->ul_pkt_cnt++;
     pdr->ul_byte_cnt += skb->len; /* length without GTP header */
-    GTP5G_INF(NULL, "PDR (%u) UL_PKT_CNT (%llu) UL_BYTE_CNT (%llu)", pdr->id, pdr->ul_pkt_cnt, pdr->ul_byte_cnt);
-
+    GTP5G_INF(NULL, "PDR (%u) UL_PKT_CNT (%llu) UL_BYTE_CNT (%llu)", pdr->id, pdr->ul_pkt_cnt, pdr->ul_byte_cnt);    
+ 
+    if (pdr->urr_num != 0) {
+        if (update_urr_counter_and_send_report(pdr, far, volume, volume_mbqe, true) < 0)
+            GTP5G_ERR(pdr->dev, "Fail to send Usage Report");
+    }
+    
+    if (color == Red) {
+        kfree_skb(skb);
```

Comment:
Maybe we need to centralize kfree_skb.
Currently some skbs are freed in inner function while others are freed in outer function 

(for example when -1 is returned ths skb is freed outside)
https://github.com/free5gc/gtp5g/blob/49fbe753afa79292ca0c09da5e640ed93dbac62d/src/gtpu/encap.c#L148C1-L151C17

Reply:
I wil check this